tracing/src/main/java/io/micronaut/tracing/brave/sender/HttpClientSenderFactory.java

```diff
@@ -46,8 +47,8 @@
      * @param invocationInstrumenterFactories The invocation instrumeter factories to instrument http client netty handlers execution with
      */
     protected HttpClientSenderFactory(
-            BraveTracerConfiguration.HttpClientSenderConfiguration configuration,
-            List<InvocationInstrumenterFactory> invocationInstrumenterFactories) {
+        BraveTracerConfiguration.HttpClientSenderConfiguration configuration,
+        List<Provider<InvocationInstrumenterFactory>> invocationInstrumenterFactories) {
```

Comment:
Given the class is public and this is protected this is a breaking change. Unfortunately its not as simple as adding a new constructor since the generic is just changing. This definitely shouldn't go into 2.4.x, however it could be argued the cost/benefit works out for 2.5. Thoughts @graemerocher ?

Reply:
I think it is highly unlikely anyone is using this API and really it should probably have been marked as `@Internal` 

That said as you say it is a breaking change although things are broken as is without this change so if we can get it into 2.5.x we should